src/main/java/com/revature/services/impl/TripServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.revature.services.impl;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.revature.models.Trip;
+import com.revature.repositories.TripRepository;
+import com.revature.services.TripService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TripServiceImpl implements TripService {
+	@Autowired
+	private TripRepository tripRepository;
+
+	@Override
+	public List<Trip> getTrips() {
+		return tripRepository.findAll();
+	}
+
+	@Override
+	public Trip getTripById(int id) {
+		return tripRepository.findById(id).get();
+	}
+
+	@Override
+	public List<Trip> getTripsByDriverId(int driverId) {
+		return tripRepository.getTripsByDriverId(driverId);
+	}
+
+	@Override
+	public List<Trip> getTripsByRiderId(int riderId) {
+		List<Integer> tripIds = tripRepository.getTripsByRiderId(riderId);
+		List<Trip> trips = null;
+		for (int r : tripIds) {
+			trips.add(tripRepository.findById(r).get());
```

Comment:
Same issue can occur here, if any of the ids do not exist in the database, an exception will be thrown.

Reply:
Refactored.